src/main/java/commission/service/PaymentCalculatorService.java

```diff
@@ -33,20 +39,25 @@ public double calculateCommissionPayment(Commission commission){
         var price = sale.get().price();
         double payment = 1;
         var percentage = 10;
-        var margin = 10;
-
-        switch (commission.commissionType()){
-            case STRAIGHT ->
-                payment = price * percentage / 100;
-            case GROSS_MARGIN ->
-                payment =  (((price * margin ) / 100 ) * 10)/100;
-            case TIERED -> {
-                if (price <= 1000){
-                    payment =   (price * 10) / 100;
-                } else if (price >= 5000 && price <10000){
-                    payment = price * 15 / 100;
-                } else if (price > 10000){
-                    payment = price * 20 / 100;
+        var margin = 40;
+
+
+        if (price == 0){
+            throw new ArithmeticException("Price cannot be 0");
```

Comment:
Nitpick: `ArithmeticException` is for mathematical errors, in your case you could throw `IllegalArgumentException` because you aren't allowing a specific value for a parameter (aka argument).

Reply:
Got it. And now I know the difference too :)

Reply:
It is good to have exceptions be specific if you want to use to control the flow of your application. Like for example when spring can't find a record in the db it returns the `EmptyResultSetDataAccessException` which is very specific